### ISSUE TITLE
Get build working

### ISSUE THREAD

**Issue by RobTillaart** [RobTillaart/SHT31_SW#2]
@ghaug

Copied the build from your latest PR to SHT31 but it fails and I don't see the difference.
Do you have time to take a look?

https://github.com/RobTillaart/SHT31_SW/actions/runs/4478285461/jobs/7870863891

Run gem install arduino_ci
  gem install arduino_ci
  ./.github/scripts/patch_arduino_ci.sh
  arduino_ci.rb
  shell: /usr/bin/bash -e {0}
Successfully installed os-1.1.4
Successfully installed rubyzip-1.3.0
Successfully installed arduino_ci-1.5.0
3 gems installed
/home/runner/work/_temp/5[2](https://github.com/RobTillaart/SHT31_SW/actions/runs/4478285461/jobs/7870863891#step:4:2)5299[3](https://github.com/RobTillaart/SHT31_SW/actions/runs/4478285461/jobs/7870863891#step:4:3)2-30c4-46b7-bb78-dfe7d85b4d03.sh: line 2: ./.github/scripts/patch_arduino_ci.sh: Permission denied
Error: Process completed with exit code 12[6](https://github.com/RobTillaart/SHT31_SW/actions/runs/4478285461/jobs/7870863891#step:4:7).

**Comment by ghaug** [comment]
The issue is that patch_arduino_ci.sh is not executable, chmod a+x on the file will fix that.

But now you lost me. Why did you copy all the code into the new library? If you make the three functions virtual in SHT31 you just need to implement these three in SHT31_SW, which is derived from SHT31.

**Comment by RobTillaart** [comment]
> The issue is that patch_arduino_ci.sh is not executable, chmod a+x on the file will fix that.

I am working under windows with a bash shell which shows the x flags set. 
(the chmod command has no effect)

```
$ ls -al
total 20
drwxr-xr-x 1 Rob 197121     0 Mar 21 12:05 ./
drwxr-xr-x 1 Rob 197121     0 Mar 21 12:05 ../
-rwxr-xr-x 1 Rob 197121 17003 Mar 21 12:05 patch_arduino_ci.sh*
```

**Comment by RobTillaart** [comment]
> But now you lost me. Why did you copy all the code into the new library?

Sorry, I had no dependency in mind. Your proposal makes sense as it would reduce the maintenance.
It means the SHT31 repo need an update / new release.

Hope to find some time slot this week to work on it.

**Comment by RobTillaart** [comment]
SHT31.h only needs
```cpp
virtual bool isConnected();
virtual bool writeCmd(uint16_t cmd);
virtual bool readBytes(uint8_t n, uint8_t *val);
```

and protected instead of private.

**Comment by ghaug** [comment]
No worries, I'm away from my desk till Friday anyway.

**Comment by RobTillaart** [comment]
The filemode disappeared when adding / pushing so some google moments later 
I found the command for setting the chmod in the add.
```
git add --chmod=+x patch_arduino_ci.sh
```

**Comment by RobTillaart** [comment]
Solved with PR #3
